programs/pkey/key_app.c

```diff
@@ -292,7 +303,7 @@ int main( int argc, char *argv[] )
 cleanup:
 
 #if defined(MBEDTLS_ERROR_C)
-    if( exit_code != MBEDTLS_EXIT_SUCCESS )
+    if( exit_code != MBEDTLS_EXIT_SUCCESS && ret != 0 )
```

Comment:
Why are both of these tested? Shouldn't these two states correlate with each other?